app/services/puppet_fact_parser.rb

```diff
@@ -166,6 +166,69 @@ def bios
     {:vendor => facts.dig('dmi', 'bios', 'vendor') || facts['bios_vendor'], :version => facts.dig('dmi', 'bios', 'version') || facts['bios_version'], :release_date => facts.dig('dmi', 'bios', 'release_date') || facts['bios_release_date']}
   end
 
+  # AWS cloud billing fields
```

Comment:
Does Puppet (or Facter) actually report this? I'd be surprised if it didn't have those as structured facts. Looking through the codebase It's not there.

It does expose EC2 data: https://github.com/puppetlabs/facter/blob/c22260225d420e72240c2b77725794041dcade6a/lib/schema/facter.yaml#L345-L366